addon/mixins/taggable-mixin.js

```diff
@@ -18,28 +18,28 @@ export default Ember.Mixin.create({
          * as the modified copy.
          *
          * @method addATag
+         * @param {DS.Model} model A model instance that supports tags functionality
          * @param {String} tag New tag to be added to list.
          */
-        addATag(tag) {
-            var resource = this.get('model');
-            var currentTags = resource.get('tags').slice(0);
+        addATag(model, tag) {
+            var currentTags = model.get('tags').slice(0);
             Ember.A(currentTags);
             currentTags.pushObject(tag);
-            resource.set('tags', currentTags);
-            return resource.save();
+            model.set('tags', currentTags);
+            return model.save();
```

Comment:
Actually, not 100% sure as to current state, with the adapter changes merged the last few days. 

These lines of code were in the original, and at the time- whether due to adapter customization or something else- the save literally didn't work at all. (I recall being particularly weirded out by line 26)